lib/grant-types/refresh-token-grant-type.js

```diff
@@ -0,0 +1,175 @@
+
+/**
+ * Module dependencies.
+ */
+
+var AbstractGrantType = require('./abstract-grant-type');
+var InvalidArgumentError = require('../errors/invalid-argument-error');
+var InvalidGrantError = require('../errors/invalid-grant-error');
+var InvalidRequestError = require('../errors/invalid-request-error');
+var Promise = require('bluebird');
+var ServerError = require('../errors/server-error');
+var is = require('../validator/is');
+var util = require('util');
+
+/**
+ * Constructor.
+ */
+
+function RefreshTokenGrantType(options) {
+  options = options || {};
+
+  if (!options.model) {
+    throw new InvalidArgumentError('Missing parameter: `model`');
+  }
+
+  if (!options.model.getRefreshToken) {
+    throw new InvalidArgumentError('Invalid argument: model does not implement `getRefreshToken()`');
+  }
+
+  if (!options.model.revokeToken) {
+    throw new InvalidArgumentError('Invalid argument: model does not implement `revokeToken()`');
+  }
+
+  if (!options.model.saveToken) {
+    throw new InvalidArgumentError('Invalid argument: model does not implement `saveToken()`');
+  }
+
+  AbstractGrantType.call(this, options);
+}
+
+/**
+ * Inherit prototype.
+ */
+
+util.inherits(RefreshTokenGrantType, AbstractGrantType);
+
+/**
+ * Handle refresh token grant.
+ *
+ * @see https://tools.ietf.org/html/rfc6749#section-6
+ */
+
+RefreshTokenGrantType.prototype.handle = function(request, client) {
+  if (!request) {
+    throw new InvalidArgumentError('Missing parameter: `request`');
+  }
+
+  if (!client) {
+    throw new InvalidArgumentError('Missing parameter: `client`');
+  }
+
+  var scope = this.getScope(request);
```

Comment:
You should not be taking scope from the user input. Refresh tokens should use the previous access token's scope because it's not guaranteed the user granted access to the scopes..

Fixed in my PR at https://github.com/night/node-oauth2-server/commit/b485b044935a9330bab959aad302994f1c3164a5